remote/nginx.conf

```diff
@@ -16,19 +16,23 @@ http {
         listen 10000;
         server_name _;
 
+        if ($cookie_token !~ '__XPRA_TOKEN__') {
+            return 401;
+        }
+
         location __XPRA_SUBPATH__ {
             rewrite ^__XPRA_SUBPATH__(.*) /$1 break;
 
-            auth_basic "Session access";
-            auth_basic_user_file /etc/nginx/.htpasswd;
-
             proxy_http_version 1.1;
             proxy_set_header Upgrade $http_upgrade;
             proxy_set_header Connection "upgrade";
             proxy_set_header Host $host;
 
             proxy_pass http://127.0.0.1:10001;
             proxy_buffering off;
+
+            proxy_hide_header Content-Security-Policy;
+            add_header Content-Security-Policy "frame-ancestors self __XPRA_CSP_ORIGIN_HOST__";
```

Comment:
I would probably add 1-2 sentences about what `__XPRA_CSP_ORIGIN_HOST__ ` is used for in the documentation, since it is neither specified/used in the Makefile nor mentioned anywhere in the documentation.